### PR TITLE
Add cpython test suites to nightly pipeline

### PR DESCRIPTION
**Pull request by vtikoo**
### Summary
- Adds cpython 3.8 test suite to nightly pipeline
- Adds handling for O_TMPFILE for ext2 and ramfs. These file systems don't support unnamed tmpfiles. Creating a tempfile will return an EISDIR  on these fs's.
- Return POLLOUT event for poll() on STDOUT_FILENO.
- Return ENOSYS for SYS_fsetxattr.
- Ext2 fixes:
    - Fixed getdents64 for files opened without O_DIRECTORY.
    - ext2_write now syncs its inode from the device before writing to it. 
    - ext2_rename: when the parent directory for old and new path are the same, link accounting for the parent directory inode had a bug.

Signed-off-by: Vikas Tikoo <vitikoo@microsoft.com>

### CODE REVIEW

File: tests/cpython-tests/Makefile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	$(APPBUILDER) -i vtikoo/cpython:3.8-ubuntu
+
+ext2fs: appdir
+	$(MYST) mkext2 appdir ext2fs
+
+clean:
+	rm -fr appdir ext2fs
+
+tests:
+	$(RUNTEST) $(MYST_EXEC) $(OPTS) ext2fs /cpython/python -m test -f pass-list -q
```
Comment (jxyang) [inline tests/cpython-tests/Makefile:25]: Please make the test file names consistent with libcxx. For example, `pass-list` -> `tests.passed`, `py-test-list` -> `tests.all`, etc.

Comment (vtikoo) [reply tests/cpython-tests/Makefile]: Changed.

File: ext2/ext2.c [hunk 2]
```diff
@@ -3354,6 +3354,12 @@ int ext2_open(
         ERAISE(-ENOTDIR);
     }
 
+    if ((flags & O_TMPFILE) && ((flags & O_RDWR) || (flags & O_WRONLY)) &&
+        S_ISDIR(locals->inode.i_mode))
+    {
+        ERAISE(-EISDIR);
```
Comment (jxyang) [inline ext2/ext2.c:3360]: Are you fixing https://github.com/deislabs/mystikos/issues/513? Can you do the same for ramfs and hostfs?

File: ext2/ext2.c [hunk 3]
```diff
@@ -5080,12 +5085,22 @@ static int _ext2_getdents64(
...
         ERAISE(-EINVAL);
 
-    if (!(file->open_flags & O_DIRECTORY))
+    if (!S_ISDIR(file->inode.i_mode))
         ERAISE(-ENOTDIR);
 
     if (count == 0)
         goto done;
 
+    if (file->dir.data == NULL)
```
Comment (jxyang) [inline ext2/ext2.c:5094]: Please add comments on when this would happen.

File: ext2/ext2.c (hunk 2 above)
Comment (vtikoo) [reply ext2/ext2.c:3360]: I have added similar check in ramfs. As hostfs can actually support tmpfiles, I didn't make any change there.

File: ext2/ext2.c (hunk 3 above)
Comment (vtikoo) [reply ext2/ext2.c:5099]: Added comment.

**Review by jxyang — APPROVED** [review]

**Pull request merged**